src/managers/input/Touch.cpp

```diff
@@ -63,7 +63,10 @@ void CInputManager::onTouchMove(wlr_touch_motion_event* e) {
 
         wlr_cursor_warp(g_pCompositor->m_sWLRCursor, nullptr, PMONITOR->vecPosition.x + e->x * PMONITOR->vecSize.x, PMONITOR->vecPosition.y + e->y * PMONITOR->vecSize.y);
 
-        const auto local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchSurfaceOrigin;
+        auto local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchSurfaceOrigin;
+        if(m_sTouchData.touchFocusWindow->m_bIsX11) {
+            local = (g_pInputManager->getMouseCoordsInternal() * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy) - m_sTouchData.touchSurfaceOrigin;
```

Comment:
(coords - windowpos) * scale - origin

Reply:
The original code is : 

```
const auto local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchSurfaceOrigin;
```

I changed it to take into account the case where the window is X11 : 

```
 auto local = g_pInputManager->getMouseCoordsInternal() - m_sTouchData.touchSurfaceOrigin;
 if (m_sTouchData.touchFocusWindow->m_bIsX11) {
      local = (g_pInputManager->getMouseCoordsInternal() * m_sTouchData.touchFocusWindow->m_fX11SurfaceScaledBy) - m_sTouchData.touchSurfaceOrigin;
}
```

All I do is to scale anew `getMouseCoordsInternal()` to X11 units. The origin has already been scaled to these units in `onTouchDown`. This could be refactored to avoid computing `local` twice, but once again I wanted to introduce as few changes as possible.

The formula would therefore pass from `coords - origin` to `coords * scale - origin`.

I tested this also and it works as expected, correcting the bug.

Should I introduce the `- windowpos` to have the formula `(coords - windowpos) * scale - origin` ? The original code does not do it, so I'm unsure, but I do not think so. This is already "included" in the `origin` calculation, and indeed not done in the original code. 

Furthermore, when introducing it, the tests fail as soon as the X11 window is not in the top left corner of the screen. My tests are done using Krita and zooming/moving the drawing with touch, and also by running `xev` that report X events and coordinates. 

I am running Hyprland with these corrections for three days now and have encountered no problems with X11 apps.

Reply:
There are no X11 units here. Everything is surface-local. This will completely implode when your monitor is not at (0,0)

Reply:
Again, x11 "units" was just a way to express the fact there are, in mathematical terms, two coordinate spaces with different metrics. Sorry for being unclear.

Reply:
but your conversions were incorrect :)

Reply:
Ha ha! NO! :P 

They are equivalent to your last code and good from the start. Your new patch code is strictly what I wrote if the rules of algebra are still valid. Substitute and solve to be convinced :)

Reply:
trust me, they aren't :P